base/essentials.jl

```diff
@@ -216,10 +216,10 @@ Evaluate an expression with values interpolated into it using `eval`.
 If two arguments are provided, the first is the module to evaluate in.
 """
 macro eval(ex)
-    :(Core.eval($__module__, $(Expr(:quote,ex))))
+    :(Core.eval($__module__, $(Expr(:escape, Expr(:quote,ex)))))
```

Comment:
Would this be simpler:
```suggestion
    return esc(Expr(:call, GlobalRef(Core, :eval), __module__, Expr(:quote, ex)))
```